frontend/src/app/main/ui/shapes/shape.cljs

```diff
@@ -54,25 +55,30 @@
         pointer-events   (unchecked-get props "pointer-events")
         disable-shadows? (unchecked-get props "disable-shadows?")
 
+        preview-blend-mode-ref
+        (mf/use-memo (mf/deps (:id shape)) #(refs/preview-blend-modes-by-id (:id shape)))
```

Comment:
- prefer use `with-memo` instead use-memo,
- extract `shape-id` as separate binding on let and then use it (for avoid calling `(:id shape)`  twice
- replace all usage of `(:id shape)` on the rest of the shape, the `(:id shape)` is a Trie Lookup operation and for performance reasons is better lookup once and then reuse